kymatio/scattering2d/frontend/torch_frontend.py

```diff
@@ -0,0 +1,96 @@
+
+        Scattering2DBase._instantiate_backend(self, 'kymatio.scattering2d.backend.')
+        Scattering2DBase.build(self)
+        Scattering2DBase.create_filters(self)
+
+        self.register_filters()
+
+    def register_filters(self):
+        """ This function run the filterbank function that
+            will create the filters as numpy array, and then, it
+            saves those arrays as module's buffers."""
+
+        # Create the filters
+        n = 0
+        for c, phi in self.phi.items():
+            if isinstance(c, int):
+                self.phi[c] = torch.from_numpy(self.phi[c]).unsqueeze(-1) # add a trailing singleton dimension to mark
+                # it as non-complex
+                self.register_buffer('tensor' + str(n), self.phi[c])
+                n += 1
+
+        for j in range(len(self.psi)):
+            for k, v in self.psi[j].items():
+                if isinstance(k, int):
+                    self.psi[j][k] = torch.from_numpy(v).unsqueeze(-1) # add a trailing singleton dimension to mark it
+                    # as non-complex
+                    self.register_buffer('tensor' + str(n), self.psi[j][k])
+                    n += 1
+
+    def scattering(self, input):
+        """Forward pass of the scattering.
+
+                Parameters
+                ----------
+                input : tensor
+                   Tensor with k+2 dimensions :math:`(n_1, ..., n_k, M, N)` where :math:`(n_1, ...,n_k)` is
+                   arbitrary. Currently, k=2 is hardcoded. :math:`n_1` typically is the batch size, whereas
+                    :math:`n_2` is the number of
+                   input channels.
+
+                Returns
+                -------
+                S : tensor
+                   Scattering of the input, a tensor with k+3 dimensions :math:`(n_1, ...,n_k, D, Md, Nd)`
+                   where :math:`D` corresponds to a new channel dimension and :math:`(Md, Nd)` are
+                   downsampled sizes by a factor :math:`2^J`. Currently, k=2 is hardcoded.
+
+                """
+        if not torch.is_tensor(input):
+            raise TypeError('The input should be a PyTorch Tensor.')
+
+        if len(input.shape) < 2:
+            raise RuntimeError('Input tensor must have at least two dimensions.')
+
+        if not input.is_contiguous():
+            raise RuntimeError('Tensor must be contiguous!')
+
+        if (input.shape[-1] != self.N or input.shape[-2] != self.M) and not self.pre_pad:
+            raise RuntimeError('Tensor must be of spatial size (%i,%i)!' % (self.M, self.N))
+
+        if (input.shape[-1] != self.N_padded or input.shape[-2] != self.M_padded) and self.pre_pad:
+            raise RuntimeError('Padded tensor must be of spatial size (%i,%i)!' % (self.M_padded, self.N_padded))
+
+        # each time scattering is run, one needs to make sure self.psi and self.phi point to
+        # the correct buffers
+        n = 0
```

Comment:
"Maybe these buffers can be retrieved in a separate function, a sort of inverse to register_filters?" Comment by @janden .